cluster-api/cloud/google/machineactuator.go

```diff
@@ -153,20 +164,41 @@ func (gce *GCEClient) Create(cluster *clusterv1.Cluster, machine *clusterv1.Mach
 		return errors.New("invalid master configuration: missing Machine.Spec.Versions.Kubelet")
 	}
 
-	image, preloaded := gce.getImage(machine, config)
+	if gce.configWatch == nil {
+		return errors.New("invalid machine setup configuration: missing GCEClient.ConfigWatch")
+	}
+	machineSetupConfigs, err := gce.configWatch.ValidConfigs()
+	if err != nil {
+		return err
+	}
+	configParams := &machinesetup.ConfigParams{
+		OS:       config.OS,
+		Roles:    machine.Spec.Roles,
+		Versions: machine.Spec.Versions,
+	}
+	image, err := machineSetupConfigs.GetImage(configParams)
+	if err != nil {
+		return err
+	}
+	imagePath := gce.getImagePath(image, config.Project)
 
+	machineSetupMetadata, err := machineSetupConfigs.GetMetadata(configParams)
+	if err != nil {
+		return err
+	}
 	if util.IsMaster(machine) {
 		if machine.Spec.Versions.ControlPlane == "" {
 			return gce.handleMachineError(machine, apierrors.InvalidMachineConfiguration(
 				"invalid master configuration: missing Machine.Spec.Versions.ControlPlane"))
 		}
 		var err error
 		metadata, err = masterMetadata(
-			templateParams{
-				Token:     gce.kubeadmToken,
-				Cluster:   cluster,
-				Machine:   machine,
-				Preloaded: preloaded,
+			metadataParams{
```

Comment:
Can we switch to just passing arguments directly as parameters. I think it would be better for the function to be explicit about which fields of this struct it's going to need.